### ISSUE TITLE
DHF 5.0.2: some write failures do not get tallied in results

### ISSUE THREAD

**Issue by crism** [marklogic/marklogic-data-hub#2895]
#### The issue

Short description of the problem: When conflicting update errors are generated during a mastering step, the step summary reports 0 failures.

What behavior are you expecting? The number of failed events should include attempts to write that failed.

```
Output:
{
    "user": "flow-developer",
    "jobId": "da882eff-d69c-478f-a8b3-ff812320491f",
    "stepResponses": {
        "4": {
            "stepOutput": [
                "com.marklogic.client.FailedRequestException: Local message: failed to apply resource at resources/ml:runFlow: Bad Request. Server Message: XDMP-CONFLICTINGUPDATES: ...
                ... dozens of conflicting updates messages ..."
            ],
            "failedBatches": 0,
            "stepDefinitionName": "smooth-mastering",
            "failedEvents": 0,
            "jobId": "da882eff-d69c-478f-a8b3-ff812320491f",
            "status": "completed step 4",
            "successfulEvents": 21360,
            "stepName": "entity-mastering",
            "flowName": "HarmonizeEntities",
            "fullOutput": null,
            "stepEndTime": "2019-08-16T10:19:53.179725-04:00",
            "successfulBatches": 855,
            "stepStartTime": "2019-08-16T10:17:42.3253-04:00",
            "totalEvents": 2935,
            "success": true,
            "stepDefinitionType": "mastering"
        }
    },
    "flowName": "HarmonizeEntities",
    "endTime": "2019-08-16T10:19:53.213095-04:00",
    "lastCompletedStep": "4",
    "lastAttemptedStep": "4",
    "jobStatus": "finished",
    "startTime": "2019-08-16T10:17:42.278784-04:00"
}
```

On a less important note, it is counterintuitive that the total events are less than the successful events, a sign that something is going wrong in the mastering process. That might be worth flagging or rewording in the report.

#### Tech details

Which Operating System are you using? CentOS 7

Which version of MarkLogic are you using? 9.0-9

Which version of the Data Hub Framework are you using? 5.0.2

Is this a QuickStart UI bug? If so, which browser are you using? N/A

#### The devil is in the details...

If possible, rerun the command with -d for debugging output and attach the output:

Debug log contains confidential customer information, but is available upon request.

**Comment by ryanjdew** [comment]
Tracking internally with ticket https://project.marklogic.com/jira/browse/DHFPROD-3092

**Comment by ryanjdew** [comment]
Closing as the fix has been released in 5.2
